File: VocaDbModel/Helpers/RelatedSitesHelper.cs
using System.Text.RegularExpressions;

namespace VocaDb.Model.Helpers
{
	public static class RelatedSitesHelper
	{
		private static readonly Regex relatedSiteUrlRegex = new Regex(@"^https?://((utaitedb\.net)|(vocadb\.net)|(touhoudb\.com))\/((Song\/Details)|S)\/\d+",
			RegexOptions.IgnoreCase | RegexOptions.ExplicitCapture);

		public static bool IsRelatedSite(string url)
		{
			if (string.IsNullOrEmpty(url))
				return false;

			return relatedSiteUrlRegex.IsMatch(url);
		}
	}
}

File: VocaDbWeb/Scripts/Helpers/RelatedSitesHelper.ts
export default class RelatedSitesHelper {
	private static relatedSiteUrlRegex = new RegExp(
		/^https?:\/\/((utaitedb\.net)|(vocadb\.net)|(touhoudb\.com))\/((Song\/Details)|S)\/\d+/,
		'i',
	);

	public static isRelatedSite = (url?: string): boolean => {
		if (!url) return false;

		return this.relatedSiteUrlRegex.test(url);
	};
}
